src/test/java/org/terasology/minecarts/RailsTest.java

```diff
@@ -40,42 +44,58 @@
 
 @ExtendWith(MTEExtension.class)
 @Dependencies({"Rails", "CoreAssets"})
-@Disabled("Failing test with Reflection")
+@Tag("MteTest")
 public class RailsTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(RailsTest.class);
+
     private static final String RAIL_BLOCKFAMILY_URI = "rails:rails";
+
     private Block dirtBlock;
     private Block airBlock;
     private BlockFamily railBlockFamily;
 
-    @In
-    BlockEntityRegistry blockEntityRegistry;
+    private boolean initialized = false;
+
     @In
     BlockManager blockManager;
     @In
     ModuleTestingHelper helper;
     @In
     WorldProvider worldProvider;
 
-    public void initialize() {
+    public void setup() {
+        logger.info("Setup blocks and block families (should only happen once)");
         airBlock = blockManager.getBlock("engine:air");
         dirtBlock = blockManager.getBlock("CoreAssets:Dirt");
+        logger.info("✓ successfully resolved blocks");
         railBlockFamily = blockManager.getBlockFamily(RAIL_BLOCKFAMILY_URI);
+        logger.info("✓ successfully resolved block family");
+    }
 
-        BlockRegion region = new BlockRegion(0, 0, 0).expand(5, 5, 5);
+    @BeforeEach
+    public void initialize() {
+        logger.info("Pre-test initialization");
+        if (!initialized) {
+            initialized = true;
+            setup();
+        }
 
-        for (Vector3ic pos : region) {
+        logger.info("Resetting test area ...");
+        for (Vector3ic pos : new BlockRegion(0, 0, 0).expand(5, 5, 5)) {
             helper.forceAndWaitForGeneration(pos);
             worldProvider.setBlock(pos, airBlock);
         }
-        for (Vector3ic pos : region) {
+        logger.info("✓ removed blocks from test space (replace with air)");
+        for (Vector3ic pos : new BlockRegion(0, -1, 0).expand(5, 0, 5)) {
             helper.forceAndWaitForGeneration(pos);
             worldProvider.setBlock(pos, dirtBlock);
         }
+        logger.info("✓ create dirt platform (place dirt blocks)");
     }
 
     @Test
     public void singleRail() {
-        this.initialize();
```

Comment:
big :+1: for swapping this from being called from each test to a `@BeforeEach` function